submissions/jelz/src/jelz/client.cljs

```diff
@@ -0,0 +1,33 @@
+(ns jelz.client
+  (:require-macros
+    [cljs.core.async.macros :refer [go go-loop]])
+  (:require
+    [chord.client :refer [ws-ch]]
+    [cljs.core.async :refer [<!]]
+    [ajax.core :refer [GET]]
+    [re-frame.core :refer [dispatch]]))
+
+(def endpoints {:planet "ws://localhost:4000"
+                :jedi   "http://localhost:3000/dark-jedis/"})
+
+(defonce ws (atom {:connection (ws-ch (:planet endpoints) {:format :json-kw})
+                   :listening? false}))
+
+(defn listen [ch]
+  (go-loop [] (let [{:keys [message]} (<! ch)]
+                (dispatch [:planet-change message])
+                (when message (recur)))))
+
+(defn setup-planet-listener []
+  (when-not (:listening? @ws)
+    (go (let [{:keys [ws-channel]} (<! (:connection @ws))]
+          (listen ws-channel)
+          (swap! ws (fn [ws] (assoc ws :listening? true)))))))
+
+(defn fetch-dark-jedi [req-id id]
+  (let [url     (str (:jedi endpoints) id)
+        handler (fn [res] (dispatch [:jedi-loaded req-id res]))
+        xhr     (GET url {:response-format :json
+                          :keywords?       true
+                          :handler         handler})]
```

Comment:
I'd be tempted to replace  use of `handler` var with inline version:
   `:handler   #(dispatch [:jedi-loaded req-id %])`       
Seems terser and more direct to me.